src/node/chainstate.cpp

```diff
@@ -221,7 +221,7 @@ ChainstateLoadResult LoadChainstate(ChainstateManager& chainman, const CacheSize
 
         // A reload of the block index is required to recompute setBlockIndexCandidates
         // for the fully validated chainstate.
-        chainman.ActiveChainstate().UnloadBlockIndex();
+        chainman.ActiveChainstate().ClearBlockIndexCandidates();
```

Comment:
It doesn't seem necessary to me, because the block arrival information is something that seems unrelated to setting up a chainstate (and I don't see why it would need to be reset in the event of a chainstate activation).

I think if we need to reset the counters it would be for tests, which is the only place outside of `init.cpp` where we use this...  I thought I addressed adding `ResetBlockSequenceCounters()` where it was necessary, but I should look at that again to see if I missed a case.